### ISSUE TITLE
In Android, call CloseVideo() then next time video not play

### ISSUE THREAD

**Issue by crazylulu9999** [RenderHeads/UnityPlugin-AVProVideo#13]
**Describe the bug**
In Android, call CloseVideo() then next time video not play. (only sound play)

Using ApplyToTextureWidgetNGUI, I found that "_mediaPlayer.TextureProducer.GetTexture()" return null in TryUpdateTexture().

That's why I call CloseVideo(), Some android devices can play only 1 video at once. (If no CloseVideo() then crash occurs.)

**Your Setup (please complete the following information):**
 - Unity version: 2018.2.20f1
 - AVPro Video version: 1.9.12
 - Operating system version: Android
 - Device model: Any
 - Video specs (resolution, frame-rate, codec, file size): Any

**To Reproduce**
1. On OnDisable(), Call CloseVideo()
2. On OnEnable(), Call OpenVideoFromFile()

**Logs**
If applicable, add error logs to help explain your problem.

**Screenshots**

**Videos**
If applicable, add a copy of your video or the URL

**Comment by krstns** [comment]
We have the exact same issue with CloseVideo() on Android.
Normally, we can call it as many times as we want CloseVideo and OpenVideoFromFile on Android without having any issues.
But as soon as we will:
- play video
- close video
- disable parent object
- enable parent object
- open video
- play video

We get a situation where we can see the preview texture, we can hear the audio, but no video is shown. I can see in logs that the video IS played - but it is covered by the preview texture.

Any tips on this one?

**Comment by AndrewRH** [comment]
Thanks for reporting the issue.  Interesting that it might be the active/enable that causes it.  We will add this to our tests and try to replicate.

**Comment by crazylulu9999** [comment]
I solved myself.

First, "OnDisable() and OnEnable()" means other script's function.

I add Dispose() into MediaPlayer.cs

```csharp
public void Dispose ()
{
	if (m_Dispose != null)
	{
		m_Dispose.Dispose();
		m_Dispose = null;
	}
	m_Control = null;
	m_Texture = null;
	m_Info = null;
	m_Player = null;

	if (m_Resampler != null)
	{
		m_Resampler.Release();
		m_Resampler = null;
	}
}
```

Call Dispose() after CloseVideo() like this.

```csharp
void StopAndCloseVideo (MediaPlayer mediaPlayer)
{
	if (mediaPlayer && (mediaPlayer.Control != null) && mediaPlayer.Control.IsPlaying())
	{
		mediaPlayer.Stop();
	}

	if (mediaPlayer && mediaPlayer.VideoOpened)
	{
		mediaPlayer.CloseVideo();
		mediaPlayer.Dispose();
		mediaPlayer.m_VideoPath = string.Empty;
	}
}
```

Then next play has no problems.

**Comment by AndrewRH** [comment]
Very cool.  We're still investigating this but thanks for the help.  You can expect this to be solved in a near future version.  Will keep this ticket open until we've implemented a solution in the plugin.

Thanks,

**Comment by AndrewRH** [comment]
Also worth noting...another user reported that he fixed this issue by adding a couple of frame-waits between video loads:

`
IEnumerator VideoPlayCoroutine()
{
mediaPlayer.Control.CloseVideo();
yield return null;
mediaPlayer.Control.CloseVideo();
yield return null;
mediaPlayer.OpenVideoFromFile(MediaPlayer.FileLocation.AbsolutePathOrURL, "https://filest.edunet.net/360vr/TotalAppAssetBundle/e161002015/Video/e161002015_Clip_1620x1022_video003.mp4");
mediaPlayer.Control.Rewind();
mediaPlayer.Control.Play();
yield return new WaitUntil(() => mediaPlayer.Control.IsFinished() == true);
mediaPlayer.Control.Stop();
}`

**Comment by RenderHeadsSte** [comment]
We have just released Version 1.9.17, which contains some fixes for issues manifested from back-to-back video open/closing in the ExoPlayer API path.  Maybe you could grab this version and check to see if the problem you are seeing has been addressed?

**Comment by AndrewRH** [comment]
We'll close this issue in a couple of days, as we've proposed a solution (new plugin version) but haven't heard anything back.  If you're still having this issue with the new version please do let us know.

Thanks,
